### PR TITLE
Migrate travis to gha

### PR DESCRIPTION
**Pull request by gchazot**
Migrate from Travis CI to Github Actions.

Also migrate the releasing workflow to use GitHub Releases as the trigger.

### CODE REVIEW

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on:
+  release:
+    types: [published]
```
Comment (DilaraOflaz) [inline .github/workflows/publish_pypi.yml:3]: How do we trigger `published` ?

**Review by DilaraOflaz — APPROVED** [review]
I've checked the Github Action docs to understand the changes
The PR LGTM

**Review by myalamov — APPROVED** [review]
Looks good to me

File: .github/workflows/publish_pypi.yml (hunk 1 above)
Comment (gchazot) [reply .github/workflows/publish_pypi.yml:3]: it's in the README

**Comment by gchazot** [comment]
Thanks for the reviews @DilaraOflaz and @myalamov

**Pull request merged**